lib/chef/chef_fs/chef_fs_data_store.rb

```diff
@@ -550,11 +585,17 @@ def _to_chef_fs_path(path)
           if path.length >= 3
             path[2] = "#{path[2]}.json"
           end
+
         elsif path[0] == "policies"
           path = path.dup
-          if path.length >= 3
-            path[2] = "#{path[2]}.json"
+          if path[2] == "revisions"
+            # Get rid of "revisions"
+            path.delete_at(2)
+            if path.length >= 3
+              path[2] = "#{path[2]}.json"
```

Comment:
What's the case here for `path > 3`? Won't that maybe leave a path like `/policies/NAME/1.2.3.json/blah`?